File: library/src/main/java/name/zeno/android/presenter/items/PoiItem.java
package name.zeno.android.presenter.items;

import android.view.View;
import android.widget.TextView;

import com.baidu.mapapi.search.core.PoiInfo;

import butterknife.BindView;
import butterknife.ButterKnife;
import kale.adapter.item.AdapterItem;
import name.zeno.android.listener.Action1;
import name.zeno.android.util.R;
import name.zeno.android.util.R2;

/**
 * @author 陈治谋 (513500085@qq.com)
 * @since 2016/12/14.
 */
public class PoiItem implements AdapterItem<PoiInfo>
{
  @BindView(R2.id.tv_poi_title)   TextView tvPoiTitle;
  @BindView(R2.id.tv_poi_address) TextView tvPoiAddress;

  Action1<PoiInfo> onClick;

  private PoiInfo poiInfo;

  @Override public int getLayoutResId()
  {

    return R.layout.item_poi_info;
  }

  @Override public void bindViews(View root)
  {
    ButterKnife.bind(this, root);
    root.setOnClickListener(view -> {
      if (onClick != null) {
        onClick.call(poiInfo);
      }
    });
  }

  @Override public void setViews()
  {

  }

  @Override public void handleData(PoiInfo poiInfo, int position)
  {
    this.poiInfo = poiInfo;
    tvPoiTitle.setText(this.poiInfo.name);
    tvPoiAddress.setText(this.poiInfo.address);
  }

  public void setOnClick(Action1<PoiInfo> onClick)
  {this.onClick = onClick; }
}

File: library/src/main/java/name/zeno/android/presenter/items/PoiItem.kt
package name.zeno.android.presenter.items

import android.view.View
import com.baidu.mapapi.search.core.PoiInfo
import kale.adapter.item.AdapterItem
import kotlinx.android.synthetic.main.item_poi_info.view.*
import name.zeno.android.util.R

/**
 * @author 陈治谋 (513500085@qq.com)
 * @since 2016/12/14.
 */
class PoiItem(
    internal var onClick: ((PoiInfo) -> Unit)
) : AdapterItem<PoiInfo> {


  private lateinit var root: View
  private lateinit var data: PoiInfo

  override val layoutResId: Int = R.layout.item_poi_info

  override fun bindViews(root: View) {
    this.root = root
    root.setOnClickListener { onClick.invoke(data) }
  }

  override fun setViews() {

  }

  override fun handleData(poiInfo: PoiInfo, position: Int) = with(root) {
    data = poiInfo
    tv_poi_title.text = data.name
    tv_poi_address.text = data.address
  }
}
